aptos-move/aptos-aggregator/src/delayed_field_extension.rs

```diff
@@ -199,18 +200,32 @@ impl DelayedFieldData {
             },
         };
 
-        let snapshot_id = resolver.generate_delayed_field_id();
+        let snapshot_id = resolver.generate_delayed_field_id(width);
         self.delayed_fields.insert(snapshot_id, change);
         Ok(snapshot_id)
     }
 
     pub fn create_new_snapshot(
         &mut self,
-        value: SnapshotValue,
+        value: u128,
+        width: usize,
```

Comment:
When value is u128, isn't width always 16 bytes?

Reply:
for code simplicity, for all integers we use u128 value internally, both for <u64> and <u128> in move. so this is needed here.